### PR TITLE
Add an action to upgrade all dependencies to the latest version, including new major version

### PR DESCRIPTION
**Pull request by ahoppen**
I wanted to put this out here for some initial feedback. Let me know if this would require an evolution proposal.

Depends on https://github.com/swiftlang/swift-syntax/pull/3366

--- 

When maintaining a non-library package, like a service executable, you may sometimes want to update all your dependencies to the latest version, including new major versions. This is particularly common when depending on packages that have a `0.x` version. Without an explicit update of your dependency requirement, you may easily be stuck on an older version of the package and not pick up bug fixes that are, eg. included in `0.12.0` while you are still on `0.10.0`.

Currently, this is a very manual process: For every package dependency, you need to eg. open the package in GitHub, check if the package has a new release and, if so, update the `from:` clause in the package manifest accordingly.

This adds a new `swift package upgrade-dependencies` action, which updates the package manifest to reference all the lastest versions. If these new versions introduce API breakages, it is expected to be the user’s responsibilities to update their package accordingly.

### CODE REVIEW

File: Tests/CommandsTests/PackageCommandTests.swift [hunk 1]
```diff
@@ -2594,6 +2596,183 @@ struct PackageCommandTests {
...
+
+                try GitRepository(path: fixturePath.appending("Foo")).tag(name: "2.0.0")
+
+                // Smoke check that the package doesn't contain a 2.0.0 dependency initially.
+                try expectManifest(packageRoot) { contents in
+                    #expect(!contents.contains(#"from: "2.0.0""#))
+                }
+
+                _ = try await execute(
+                    ["upgrade-dependencies"],
```
Comment (bkhouri) [inline Tests/CommandsTests/PackageCommandTests.swift]: Suggestion (blocking): Can we add additional automated tests that verify other scenarios?  At a minimum, we should have a test that executes `upgrade-dependencies`:

- where the directory tree does not contains a `Package.swift`.  I believe this should fail with `unknown package`
- where the `Package.swift` file is not readable to the current user. I believe this should fail with `cannot find package manifest in <path-to-Package.swift>`
- where 0 package depedencies have newer versions.  We expect the `Package.swift` to be unchanged.
- where 2 packages dependencies have newer versions
- where 3+ (multiple) packages dependencies have newer versions
- where `Package.swift` contains package dependencies that use a package registry. Executes the same number of tests where we validate a package dependency on a git URL.

**Review by bkhouri — CHANGES_REQUESTED** [review]
Thanks for this PR @ahoppen.  I am marking this a "request change" as I feel we can add additional automated tests to ensure the feature behaves as we expect in different use cases.

File: Tests/CommandsTests/PackageCommandTests.swift (hunk 1 above)
Comment (ahoppen) [reply Tests/CommandsTests/PackageCommandTests.swift]: Added the tests and also added support for resolving the latest version from a package registry. Thanks for that suggestion, I forgot about that.

> * where 3+ (multiple) packages dependencies have newer versions

I skipped this because I don’t think there’s any significant difference between 2 and 3 updated package dependencies.

Comment (bkhouri) [reply Tests/CommandsTests/PackageCommandTests.swift]: >I skipped this because I don’t think there’s any significant difference between 2 and 3 updated package dependencies.

I usually try the 0, 1, 2 and many.  In this case, 3 covers the many use case.
